### PR TITLE
chore: adding workflow to add issues to docs on docs needed label

### PR DESCRIPTION
**Pull request by signorecello**
# Description

## Summary of changes

It creates a new workflow that automatically creates new issues in the docs repo when someone assigns the "doc needed" label here

# Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.

### CODE REVIEW

**Comment by signorecello** [comment]
done @TomAFrench

File: .github/workflows/label_docs.yml
Comment (TomAFrench) [inline .github/workflows/label_docs.yml:29]: We already have an established way to trigger reusable workflows which we use for triggering releases:

https://github.com/noir-lang/noir/blob/73df4653556a7d1c74d184e27ec5a8ca3be47af9/.github/workflows/release.yml#L66-L73

Could you update to use this?

**Review by TomAFrench — CHANGES_REQUESTED** [review]
We don't need two copies of the `dispatch_open` and `dispatch_close` job afaict. We only need to change the workflow id between them so we should just need an "if-else" on that argument.

File: .github/workflows/label_docs.yml
Comment (phated) [inline .github/workflows/label_docs.yml:31]:
```suggestion
          token: ${{ secrets.DOCS_REPO_TOKEN }} 
```
I believe we need a token with access to the docs repo added to the secrets

File: .github/workflows/label_docs.yml
Comment (phated) [inline .github/workflows/label_docs.yml:30]: Small nit: @kevaundray wanted to standardize all default branches to `master`

File: .github/workflows/label_docs.yml
Comment (kevaundray) [reply .github/workflows/label_docs.yml:31]: Added this to the repo secrets so committing

File: .github/workflows/label_docs.yml
Comment (kevaundray) [reply .github/workflows/label_docs.yml:30]: Thats a good catch -- I've just set the default to master since most of our repositories should already use this

Comment (signorecello) [reply .github/workflows/label_docs.yml:30]: I'll change the branch name to "master" on the docs repo

Comment (signorecello) [reply .github/workflows/label_docs.yml:30]: done

**Comment by kevaundray** [comment]
Leaving this to @phated as I don't know enough about the CI

File: .github/workflows/label_docs.yml
Comment (phated) [inline .github/workflows/label_docs.yml:1]:
```suggestion
name: Notify Doc Needed
```

**Review by phated — DISMISSED** [review]
LGTM!

**Comment by signorecello** [comment]
@TomAFrench can you review this so it gets merged?

**Review by TomAFrench — APPROVED** [review]

**Comment by TomAFrench** [comment]
Actually, @signorecello could we merge the workflows that this workflow will be triggering first before we merge this? Not sure how GH will react to a missing workflow.

**Comment by TomAFrench** [comment]
Holding off for a little bit due to https://github.com/noir-lang/noir/pull/1193#issuecomment-1517770174

**Review by kevaundray — APPROVED** [review]

**Pull request merged**
